src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContext.cs

```diff
@@ -74,6 +75,7 @@ public DependencyContext Merge(DependencyContext other)
                 );
         }
 
+        [RequiresAssemblyFiles(Message = "Calls GetDepsJsonPath")]
```

Comment:
This should be propagated to the public API (Default property) - I would expect the analyzer to generate some warnings around this. If not it's a bug in the analyzer which we should fix.

Reply:
I tried a little bit today to debug this, seems like a more complicated scenario than I expected. The analyzer is only able to see System.Lazy<DependencyContext>.Value.get but I have not found a way to connect it with the function is calling (LoadDefault). Also, I tested a similar scenario to see how does the linker behaves using RequiresUnreferencedCode and I found that they both have the same testing hole and don't produce the warning.
```C#
public static void Main ()
{
  TestCallsLazyInitializer ();
}

public static Lazy<RequiresUnreferencedCodeCapability> lazyRUC = new Lazy<RequiresUnreferencedCodeCapability> (InitRUC);

[RequiresUnreferencedCode ("Message for --RequiresUnreferencedCodeWithLazyInit--")]
public static RequiresUnreferencedCodeCapability InitRUC ()
{
	RequiresUnreferencedCodeCapability objectRUC = new RequiresUnreferencedCodeCapability ();
	return objectRUC;
}

[ExpectedWarning ("IL2026", "--RequiresUnreferencedCodeWithLazyInit--")]
public static void TestCallsLazyInitializer ()
{
	RequiresUnreferencedCodeCapability objectRUC = lazyRUC.Value;
}
```

Reply:
That looks like https://github.com/mono/linker/issues/1912 and related.

We should get a warning in the static constructor, when the Lazy instance is getting constructed.

Reply:
With https://github.com/mono/linker/pull/1964 now the Lazy call showed a warning, but since it warns on an implicit static constructor (I cannot annotate the static constructor unless I implement it) and then gets assigned to a field (which is a target that RequiresAssembly files don't support) I couldn't mark it with RequiresAssemblyFiles. I ended up refactoring the code and not make use of the Lazy call.